### DISCUSSION TITLE
Notcurses as a final display output to a generic renderer

### DISCUSSION THREAD

**Discussion by l1mey112** [dankamongmen/notcurses#2667] [Q&A] [answered]
I want to write directly to the screen buffer, that can change size at any time. If this is possible I would need a way to get the screen's dimensions beforehand so I can adjust the final images size.

If this is not possible, is there a way to create a simple image to write to and after finishing, push that to the screen? If you get back to me, a simple example will be greatly appreciated!

Maybe something similar to this SDL C code?
```c
...
// edit an allocated buffer with colour data
(Uint8*)(m_window_surface->pixels)[offset + 0] = col.b;
(Uint8*)(m_window_surface->pixels)[offset + 1] = col.g;
(Uint8*)(m_window_surface->pixels)[offset + 2] = col.r;
...

// update the window with the buffer
SDL_UpdateWindowSurface(m_window);
```

I want to integrate notcurses as the display output from an opengl compute shader. 
Thank you for taking the time to look over my question.

\- lm

**Comment by dankamongmen** [comment]
awesome! i'm about to step away from my computer, but i'll try to reply to this tonight -- feel free to poke me if i haven't =]

**Comment by l1mey112** [comment]
Any updates?

**Comment by l1mey112** [comment, marked as answer]
https://github.com/l1mey112/notcurses-opengl
